[bachelor] 修改格式，使其更符合学士学位论文模板要求

chenlheng opened the pull request:
1. 上传目前学士学位论文格式要求文件，并根据要求内容做了以下修改：
- 封面英文标题：THESIS OF BACHELOR -> BACHELOR'S THESIS
- 页眉：黑色banner -> 红色banner（使用的是与格式要求相同的暗红色，而非120周年校庆后视觉标准里的鲜红色）
- 页眉：标题文字字体：楷书 -> 黑体
- 章节标题缩进：section和subsection增加2字符长度缩进
- 章节标题行间距：chapter, section和subsection分别改为前后各1.0\baselineskip, 0.5\baselineskip, 0\baselineskip（由于未用到subsubsection，故未做修改）
- 图表标题：font增加bf，中文改成heiti
2. 针对编译过程中出现的参考文献丢失，在thesis.tex内增加解决方案，并在faq内增加相关QA
3. 针对查重过程出现的中文字符无法识别问题，在faq内增加相关QA

chenlheng committed:
Update

1. fix format of bachelor's thesis and upload an official requirement
2. fix a bug with bibliography
3. update faqs on bibliography and Chinese recognition

gaocegege commented:
谢谢你的贡献 :smile:

gaocegege commented on tex/faq.tex:
```diff
@@ -25,6 +25,10 @@ \chapter{常见问题}
 
+{\bfseries{}Q:为什么无法生成参考文献页，且正文中的引用全部变成了引用序号上标全部变成了引用关键字了？}
+
+A: 出现这种情况请先确认自己使用的bibliography tool是Biber，再在thesis.tex内在数据库导入部分注释掉\verb|\addbibresource{}|，并取消\verb|\bibliography{}|的注释。
```
中英文之间请加上空格

gaocegege commented on thesis.tex:
```diff
@@ -4,8 +4,8 @@
-\documentclass[doctor, openright, twoside]{sjtuthesis}
-% \documentclass[bachelor, openany, oneside, submit]{sjtuthesis}
+%\documentclass[doctor, openright, twoside]{sjtuthesis}
+ \documentclass[bachelor, openany, oneside, submit]{sjtuthesis}
```
默认还请保持是 doctor

gaocegege reviewed (commented):
学士学位论文格式要求.pdf 这个文件不需要引入 SJTUThesis 吧？

gaocegege commented:
PR 修改了一些 Global 的配置，不止会影响学士学位论文模板，所以 @AlexaraWu @weijianwen PTAL

chenlheng commented:
谢谢回复，请问我需要进行修改后重新发起request吗？还是这样就可以了？

gaocegege commented:
@chenlheng 你好，你在你的 master 分支上继续提交新的 commit 改一下就好了，这里也会看到相应的 commit 的

AlexaraWu commented:
这不是学士学位论文格式要求，这是 Word 版的模版。真正的撰写规范在这里：

[本科生毕业设计（论文）撰写规范.doc - 上海交通大学教务处](http://www.jwc.sjtu.edu.cn/Upload/Content/Attach/8_634688806561790000_385339.doc)

Word 模版仅作参考， LaTeX 有自己的审美风格，没有必要照搬。

chenlheng committed:
Update

1. Remove "学士学位论文格式要求.pdf"
2. Reformat changed part in faq.tex
3. Restore to default setting in thesis.tex

chenlheng commented:
已提交新的Update:
1. Remove "学士学位论文格式要求.pdf"
2. Reformat changed part in faq.tex
3. Restore to default setting in thesis.tex

@AlexaraWu 发起PR的主要初衷是
1. 模板中存在部分瑕疵，比如封面的标题不太妥当等。
2. 实际使用过程中出现了一些意外，比如参考文献和无法识别中文的问题。
3. 包括我在内的许多人使用模板时是以学校发布的 Word 版规范为准，会花很多时间在修改格式。
因此提交论文后根据使用情况对内容进行修改，并发起PR希望能帮到后面使用这个模板的人。
如果直接照搬不太合适的话是否可以解决下前两个问题，最后一个问题在faq内进行提醒？

gaocegege commented:
@AlexaraWu @chenlheng 

可否这样，cls 文件里 global 的配置 revert 掉保持不变，而专门对 bachelor 学位论文模板的修改 merge 到 master？

AlexaraWu commented:
@chenlheng 

1. 根据《撰写规范》的要求：“毕业设计（论文）采用上海交通大学本科生毕业设计（论文）统一封面。” 学校提供的封面文件为 [本科生毕业论文内封面.doc](http://zhiyuan.sjtu.edu.cn/file/common/20171024153642_%E6%9C%AC%E7%A7%91%E7%94%9F%E6%AF%95%E4%B8%9A%E8%AE%BA%E6%96%87%E5%86%85%E5%B0%81%E9%9D%A2.doc)，其中的英文标题就是 “THESIS OF BACHELOR” 而非 “BACHELOR'S THESIS”，这里以此文件为准；

1. 《撰写规范》并未明确页眉格式，选用黑色校徽的原因是为了兼顾电子版和打印版，字体使用楷体而不是是黑体，是因为黑体在不同中文字体设置下效果差别较大；

1. 章节标题缩进我之前也设置过，但实际缩进和字号大小有关，我觉得很难看，这个待议；

1. 行间距是沿用的之前的设置我觉得不需要改；

1. 图表标题《撰写规范》里的要求是 “宋体加黑”，关于这个问题我在 #308 里叙述过，还是用楷体较好。

以上是我对格式的看法。

关于你提到的参考文献丢失和无法识别中文的问题能不能单独开 Issue 再讨论一下？

hushidong commented:
参考文献丢失是什么问题？跟addbibresourse命令有关系？

gaocegege commented:
@hushidong 

> Q:为什么无法生成参考文献页，且正文中的引用全部变成了引用序号上标全部变成了引用关键字了？
A: 出现这种情况请先确认自己使用的 bibliography tool 是 Biber，再在 thesis.tex 内在数据库导入部分注释掉\verb|\addbibresource{}|，并取消\verb|\bibliography{}|的注释。

AlexaraWu commented:
无法生成参考文献页难道不是编译流程不正确的问题吗？感觉很多第一次用的人都是只在编辑器里走了一遍 xelatex。

hushidong commented:
应该是流程问题，biblatex下之所以还能用bibliography命令是为了兼容老的习惯，但这是过时命令，一般不用。

有个示例看一下就好了，中英文判断这种问题应该也是有什么特殊情况的，@chenlheng  ?

chenlheng commented:
@AlexaraWu OK，这样的话我能理解为什么会存在这些差异了。话说要不要考虑在 faq 里面放下这部分解释？这样用的人也比较好接受这些不同，也能帮助他们决定是否要改回去。
另外两个问题可以按照我在 faq 里给出方法解决，如果需要从根本上解决的话是不是可以内部讨论下，我就不开issue了。

hushidong commented:
问题能否复现？不然不太好说清楚。

hushidong commented:
你说的会不会是没加扩展名.bib导致的？因为add命令是必须加扩展名的。而bibliography命令习惯上不加。

chenlheng commented:
@hushidong 
我尝试着复现了中文字符丢失的问题：
在本地编译 master 上最新的 commit，上传至学院提供的维普查重，结果如下：
![image](https://user-images.githubusercontent.com/26353914/41015934-0990b6c8-697f-11e8-8ea1-7875f01083dd.png)
在本地执行 make wordcount 命令：
![image](https://user-images.githubusercontent.com/26353914/41015846-acba6d72-697e-11e8-9699-cb510b58ef97.png)
直接打包上传至ShareLaTeX，选择 XeLaTeX 进行编译，生成的 pdf 文件显示正常，但查重字数统计如下：
![image](https://user-images.githubusercontent.com/26353914/41015950-1abe8844-697f-11e8-9c1e-b71d3c903c01.png)
由于我现在没有查重次数，所以无法生成检测报告。这是之前出现这种问题的同学的检测报告的目录页及图片公式索引，中文全部丢失：
![image](https://user-images.githubusercontent.com/26353914/41016059-c9c56c68-697f-11e8-9baf-3ab7193b1b2d.png)
这个可能是ShareLaTeX那边的问题，建议加入 faq 排个雷

复现过程中模板使用 \addbibresource，参考文献显示正常，这个问题暂时无法复现，如果后续有再出现这个情况我会重新报告的。不过之前出现问题时这句命令没有修改过。

hushidong commented:
好的，感谢！

gaocegege commented:
@chenlheng 

谢谢你的贡献，可以修改下这个 PR，只保留 FAQ 部分么

chenlheng committed:
Update

1. Remove changes except those related to Chinese characters recognition

chenlheng commented:
@gaocegege 已经修改了，FAQ内关于参考文献的问题也已经去掉了

chenlheng committed:
Update

1. Fix a mistake in default setting of template

gaocegege commented:
LGTM

gaocegege commented:
Thanks for your contribution!

gaocegege commented:
Fuck Travis

chenlheng commented:
hhhhhh
